pkgs/applications/science/electronics/hal-hardware-analyzer/default.nix

```diff
@@ -1,90 +1,100 @@
 { lib
 , stdenv
+, boost
+, cmake
 , fetchFromGitHub
 , fetchpatch
-, cmake
+, graphviz
+, igraph
+, llvmPackages
 , ninja
 , pkg-config
 , python3Packages
-, boost
-, rapidjson
 , qtbase
 , qtsvg
-, igraph
+, quazip
+, rapidjson
 , spdlog
+, suitesparse
 , wrapQtAppsHook
-, graphviz
-, llvmPackages
 , z3
-, fmt_8
-, suitesparse
 }:
 
 let
-  igraph' = igraph.overrideAttrs (old: rec {
+  igraph' = igraph.overrideAttrs (final: prev: {
     version = "0.9.10";
     src = fetchFromGitHub {
       owner = "igraph";
-      repo = "igraph";
-      rev = version;
+      repo = final.pname;
+      rev = final.version;
       hash = "sha256-prDadHsNhDRkNp1i0niKIYxE0g85Zs0ngvUy6uK8evk=";
     };
-    postPatch = old.postPatch + lib.optionalString stdenv.isAarch64 ''
+    patches = (prev.patches or []) ++ [
+      # needed by clang
+      (fetchpatch {
+        name = "libxml2-2.11-compat.patch";
+        url = "https://github.com/igraph/igraph/commit/5ad464be5ae2f6ebb69c97cb0140c800cc8d97d6.patch";
+        hash = "sha256-adU5SctH+H54UaAmr5BZInytD3wjUzLtQbCwngAWs4o=";
+      })
+    ];
+    postPatch = prev.postPatch + lib.optionalString stdenv.isAarch64 ''
       # https://github.com/igraph/igraph/issues/1694
       substituteInPlace tests/CMakeLists.txt \
         --replace "igraph_scg_grouping3" "" \
         --replace "igraph_scg_semiprojectors2" ""
     '';
-    buildInputs = old.buildInputs ++ [ suitesparse ];
-    cmakeFlags = old.cmakeFlags ++ [ "-DIGRAPH_USE_INTERNAL_CXSPARSE=OFF" ];
+    buildInputs = prev.buildInputs ++ [ suitesparse ];
+    cmakeFlags = prev.cmakeFlags ++ [ "-DIGRAPH_USE_INTERNAL_CXSPARSE=OFF" ];
```

Comment:
These features are specific to HAL? I also wonder whether they should be part of the main expression.

Reply:
Yeah I think these were just "brought back" from the 0.9.x package to make this overridden version as similar as possible to the old 0.9.x package.

Reply:
> Yeah I think these were just "brought back" from the 0.9.x package to make this overridden version as similar as possible to the old 0.9.x package.

It'd be nice to write that in a comment.